src/nemos/base_regressor.py

```diff
@@ -0,0 +1,465 @@
+    scoring the model, simulating responses, and preprocessing data. Concrete classes
+    are expected to provide specific implementations of the abstract methods defined here.
+    Below is a table listing the default and available solvers for each regularizer.
+
+    | Regularizer   | Default Solver   | Available Solvers                                           |
+    | ------------- | ---------------- | ----------------------------------------------------------- |
+    | UnRegularized | GradientDescent  | GradientDescent, BFGS, LBFGS, NonlinearCG, ProximalGradient, LBFGSB |
+    | Ridge         | GradientDescent  | GradientDescent, BFGS, LBFGS, NonlinearCG, ProximalGradient, LBFGSB |
+    | Lasso         | ProximalGradient | ProximalGradient                                            |
+    | GroupLasso    | ProximalGradient | ProximalGradient                                            |
+
+    Parameters
+    ----------
+    regularizer :
+        Regularization to use for model optimization. Defines the regularization scheme
+        and related parameters.
+        Default is UnRegularized regression.
+    regularizer_strength :
+        Float that is default None. Sets the regularizer strength. If a user does not pass a value, and it is needed for
+        regularization, a warning will be raised and the strength will default to 1.0.
+    solver_name :
+        Solver to use for model optimization. Defines the optimization scheme and related parameters.
+        The solver must be an appropriate match for the chosen regularizer.
+        Default is `None`. If no solver specified, one will be chosen based on the regularizer.
+        Please see table above for regularizer/optimizer pairings.
+    solver_kwargs :
+        Optional dictionary for keyword arguments that are passed to the solver when instantiated.
+        E.g. stepsize, acceleration, value_and_grad, etc.
+         See the jaxopt documentation for details on each solver's kwargs: https://jaxopt.github.io/stable/
+
+    See Also
+    --------
+    Concrete models:
+
+    - [`GLM`](../glm/#nemos.glm.GLM): A feed-forward GLM implementation.
+    - [`GLMRecurrent`](../glm/#nemos.glm.GLMRecurrent): A recurrent GLM implementation.
+    """
+
+    def __init__(
+        self,
+        regularizer: Union[str, Regularizer] = "UnRegularized",
+        regularizer_strength: Optional[float] = None,
+        solver_name: str = None,
+        solver_kwargs: Optional[dict] = None,
+    ):
+        self.regularizer = regularizer
+
+        # check regularizer strength
+        if regularizer_strength is None:
+            warnings.warn(
+                UserWarning(
+                    "Caution: regularizer strength has not been set. Defaulting to 1.0. Please see "
+                    "the documentation for best practices in setting regularization strength."
+                )
+            )
+            self._regularizer_strength = 1.0
```

Comment:
```suggestion
            self.regularizer_strength = 1.0
```
This becomes mandatory if you accept to tree_map the regularizer strength